internal/controller/drplacementcontrol.go

```diff
@@ -1186,22 +1194,48 @@ func (d *DRPCInstance) readyToSwitchOver(homeCluster string, preferredCluster st
 		}
 	}
 	// Allow switch over when PV data is ready and the cluster data is protected
-	return d.isVRGConditionMet(homeCluster, VRGConditionTypeDataReady) &&
-		d.isVRGConditionMet(homeCluster, VRGConditionTypeClusterDataProtected)
+	dataReady, err := d.isVRGConditionMet(homeCluster, VRGConditionTypeDataReady)
+	if err != nil || !dataReady {
+		return false
+	}
+
+	clusterDataReady, err := d.isVRGConditionMet(homeCluster, VRGConditionTypeClusterDataProtected)
+
+	return err == nil && clusterDataReady
 }
 
-func (d *DRPCInstance) checkReadiness(homeCluster string) bool {
+func (d *DRPCInstance) checkReadiness(homeCluster string) (bool, error) {
 	vrg := d.vrgs[homeCluster]
 	if vrg == nil {
-		return false
+		d.log.Info(fmt.Sprintf("checkReadiness: VRG not available on cluster %s", homeCluster))
+
+		return false, fmt.Errorf("VRG not available on cluster %s", homeCluster)
```

Comment:
If we return an error, why log? The caller should log this.

Reply:
Done